ui/page/transaction/transactions_page.go

```diff
@@ -371,35 +506,58 @@ func (pg *TransactionsPage) HandleUserInteractions() {
 		break
 	}
 
+	if pg.isHomepageLayout && pg.walletDropDown != nil && pg.walletDropDown.Changed() {
+		pg.selectedWallet = pg.assetWallets[pg.walletDropDown.SelectedIndex()]
+		pg.refreshAvailableTxType()
+		go pg.scroll.FetchScrollData(false, pg.ParentWindow())
+	}
+
 	if clicked, selectedItem := pg.transactionList.ItemClicked(); clicked {
 		transactions := pg.scroll.FetchedData()
-		pg.ParentNavigator().Display(NewTransactionDetailsPage(pg.Load, transactions[selectedItem], false))
+		tx, wal := pg.txAndWallet(transactions[selectedItem])
+		pg.ParentNavigator().Display(NewTransactionDetailsPage(pg.Load, wal, tx, false))
+	}
+
+	dropDownList := []*cryptomaterial.DropDown{pg.txTypeDropDown}
+	if pg.isHomepageLayout && pg.walletDropDown != nil {
+		dropDownList = append(dropDownList, pg.walletDropDown)
 	}
-	cryptomaterial.DisplayOneDropdown(pg.txTypeDropDown)
+	cryptomaterial.DisplayOneDropdown(dropDownList...)
+
+	if pg.txCategoryTab.Changed() {
+		pg.selectedTxCategoryTab = pg.txCategoryTab.SelectedIndex()
+		if pg.isHomepageLayout {
+			pg.initWalletDropdown()
+		}
 
-	if tabItemClicked, clickedTabIndex := pg.tabs.ItemClicked(); tabItemClicked {
-		pg.selectedTabIndex = clickedTabIndex
 		pg.refreshAvailableTxType()
 		go pg.scroll.FetchScrollData(false, pg.ParentWindow())
 	}
 }
 
 func (pg *TransactionsPage) listenForTxNotifications() {
+	if pg.selectedWallet == nil {
+		return
+	}
```

Comment:
Listen for ntfns for all wallets. But ignore ntfns if the wallet sending the ntfn is not the currently selected wallet. Can make this a TODO for now, then I can implement it in the PR I plan to follow this up with.